### ISSUE TITLE
Crash on Android with Mopub integration.

### ISSUE THREAD

**Issue by fourplay45** [AdColony/AdColony-Android-SDK-3#30]
I'm using the following plugin and it **crash on app start every time** for release build (with proguard config listed in your integration page.)
implementation 'com.adcolony:sdk:3.3.3'
implementation 'com.google.ads.mediation:adcolony:3.3.2.0'

```
04-17 20:08:01.482 23400-23400/? E/AdColony: JNI_OnLoad
04-17 20:08:01.498 23400-23571/? A/libc: Fatal signal 11 (SIGSEGV), code 1, fault addr 0xa in tid 23571 (pool-16-thread-), pid 23400 (indrapp.android)
04-17 20:08:01.538 23586-23586/? A/DEBUG: *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
    Build fingerprint: 'google/taimen/taimen:8.1.0/OPM2.171019.029/4657601:user/release-keys'
    Revision: 'rev_10'
    ABI: 'arm64'
    pid: 23400, tid: 23571, name: pool-16-thread-  >>> com.grindrapp.android <<<
    signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0xa
    Cause: null pointer dereference
04-17 20:08:01.539 23586-23586/? A/DEBUG:     x0   000000759c448fc8  x1   00000075b30e9e90  x2   0000000000000001  x3   0000000000000002
        x4   000000759c4490b8  x5   000000759c4490b0  x6   002c002200340038  x7   005f0073006f0022
        x8   0000000000000002  x9   0000000000000001  x10  0000000000000001  x11  00000075b30e8078
        x12  0000000000040404  x13  0000000000040000  x14  0000000000000000  x15  00000075b9000000
        x16  000000759c5df9f0  x17  000000763d47752c  x18  000000000000feff  x19  000000759c4490b0
        x20  00000075b31881d8  x21  00000075b3188150  x22  000000000000000a  x23  000000759c4490c0
        x24  0000000000000000  x25  2072c1fbb7e43d41  x26  0000000000000001  x27  000000759c448fb8
        x28  2072c1fbb7e43d41  x29  000000759c4490a0  x30  000000759c482dd8
        sp   000000759c448f90  pc   000000759c482e74  pstate 0000000080000000
04-17 20:08:01.540 23586-23586/? A/DEBUG: backtrace:
        #00 pc 0000000000037e74  /data/app/com.grindrapp.android-OwtnVkGVsC83HUfuh85ydQ==/lib/arm64/libd.so (JSObjectCallAsFunction+272)
        #01 pc 00000000000048ec  /data/app/com.grindrapp.android-OwtnVkGVsC83HUfuh85ydQ==/lib/arm64/libe.so (Java_o__01422_00024_002cb_c+696)
        #02 pc 000000000006693c  /data/app/com.grindrapp.android-OwtnVkGVsC83HUfuh85ydQ==/oat/arm64/base.odex (offset 0x66000)
04-17 20:08:02.778 859-859/? E//system/bin/tombstoned: Tombstone written to: /data/tombstones/tombstone_07
```

**Comment by fourplay45** [comment]
The version I can use it now is : "com.adcolony:sdk:3.1.2"

**Comment by perrychinn** [comment]
Hi,
Thank you for reaching out to us!  Do you get the same crashes when using debug mode (without proguard)?  It looks like your gradle also uses the AdMob mediation.  We released our SDK v3.3.3 a few days ago, so the Mopub and AdMob adapters still reference our v3.3.2.  Were you getting the same crashes using our SDK v3.3.2  too?  Thanks.

Regards,
Perry

**Comment by fourplay45** [comment]
Yes, we can't upgrade Adcolony since 3.1.2 because of the crash.
Debug mode is working fine so I'm wondering if for proguard setup do we need more then what we have it here:

```
# For communication with AdColony's WebView
-keepclassmembers class * { 
    @android.webkit.JavascriptInterface <methods>; 
}

# Keep ADCNative class members unobfuscated
-keepclassmembers class com.adcolony.sdk.ADCNative** {
    *;
 }
```

**Comment by pchinn** [comment]
resolved #30
